File: cpp-03/ex1/ft_score_analytics.cpp
#include <algorithm>
#include <iostream>
#include <vector>
#include <numeric>

int main(int argc, char** argv)
{
    std::cout << "=== Player Score Analytics ===" << std::endl;
    if (argc < 2)
        std::cout << "No scores provided. Usage: a.out <score1> <score2> ..." << std::endl;
    else
    {
        std::vector<int> score_list;
        int total_score = 0, score = 0, range = 0, max = 0, min = 0;

    while(*++argv)
        try
        {
            score_list.push_back(std::stoi(*argv));
        }
        catch (const std::invalid_argument& e)
        {
            std::cout << "Error: " << *argv << "is not a valid score!" << std::endl;
        }

    std::cout << "Scores processed: [";
    for(auto& score: score_list)
         std::cout << score << ", ";
    std::cout << "]" << std::endl;
    total_score = std::accumulate(score_list.begin(), score_list.end(), 0);
    max = *std::max_element(score_list.begin(), score_list.end());
    min = *std::min_element(score_list.begin(), score_list.end());
    std::cout << "Total players: " << argc - 1 << std::endl;
    std::cout << "Total score: " << total_score << std::endl;
    std::cout << "Average score: " << total_score / score_list.size() << std::endl;
    std::cout << "High score: " << max << std::endl;
    std::cout << "Low score: " << min << std::endl;
    if (score_list.size() == 1)
        range = max;
    else
        range = max - min;
    std::cout << "Score range: " << range << std::endl;
    }
    return 0;
}

File: python-03/ex1/ft_score_analytics.py
#!/usr/bin/env python3

import sys

if __name__ == "__main__":
    print("=== Player Score Analytics ===")

    if len(sys.argv) < 2:
        print(
            "No scores provided. Usage: python3 "
            "ft_score_analytics.py <score1> <score2> ..."
        )
    else:
        score_list = []
        for arg in sys.argv[1:]:
            try:
                score_list += [int(arg)]
            except ValueError:
                print(f"Error: {arg} is not a valid score!")

        print(f"Scores processed: {score_list}")
        total_score = sum(score_list)
        high_score = max(score_list)
        low_score = min(score_list)

        print(f"Total players: {len(sys.argv) - 1}")
        print(f"Total score: {total_score}")
        print(f"Average score: {total_score / len(score_list)}")
        print(f"High score: {high_score}")
        print(f"Low score: {low_score}")

        score_range = 0
        if len(score_list) == 1:
            score_range = high_score
        else:
            score_range = high_score - low_score
        print(f"Score range: {score_range}")
